### DISCUSSION TITLE
useFieldArray with MultiSelect component

### DISCUSSION THREAD

**Discussion by Skia69** [react-hook-form/react-hook-form#9421] [Q&A] [answered]
Hello,

Is there an example somewhere that shows how to wire up `useFieldArray` with a `MultiSelect` or `CheckboxGroup` type of component (whose its onChange will always return an array of values) in order to use the append function that `useFieldArray` exposes so that you're able to create input fields dynamically based on the number of those values?

Currently, i have a `MultiSelect` input element from which i can pick multiple values, which will in turn be appended to an array of `Attributes` through an intermediate `useState`. I'm working with this component in particular: [https://mantine.dev/core/multi-select/](url)

What i initially did was something as follows: 
```js
const value = roundAttributes.map((attr) => attr.value) // this data is coming from an api

const [attributes, setAttributes] = React.useState(value)

    const form = useForm({
        defaultValues: {
            attributes: [],
        },
    })

   <MultiSelect
          placeholder="Select one or more attributes"
          value={attributes}
          data={gameAttributes} 
          onChange={setAttributes}
    />
    
       <FormProvider {...form}>
                <form onSubmit={handleSubmit(onHandleSubmit)}>
                            {attributes?.map((attribute, index) => {
                                return (
                                    <React.Fragment key={attribute}>
                                                <AttributesForm
                                                    {...{
                                                        attribute,
                                                        index,
                                                        gameAttributes,
                                                        roundAttributes,
                                                        scrims,
                                                        isRequired,
                                                    }}
                                                />
                                    </React.Fragment>
                                )
                            })}     
                </form>
            </FormProvider>
```

The reason why i have the `MultiSelect` component outside the form is because i am using this as a `ghost` component? If that makes sense 😅

And inside the `AttributesForm` component i am dynamically creating input elements as follows:
```js
 <FormInput name={`attributes.${index}.id`} />
 <FormInput name={`attributes.${index}.points`} />
 <FormInput name={`attributes.${index}.order_number`} />
 ```

This is a heavily trimmed down version, though as you can see it is really a not so clean implementation. It works just fine but it's sort of buggy since it doesn't really remove an element from the array if i wish so.

As you can see, my implementation is far from perfect,
So what i am trying to do now is swap the whole thing with `useFieldArray` but i am struggling with connecting the dots

Any sort of help or guidance would be appreciated,
Thanks in advance.

**Comment by bluebill1049** [comment]
Did you wrap that mui-select with controller or hooked with useController?

**Reply by Skia69** [reply]
In my particular use case i didn't see a need for that. Though if you check @Phibee sandbox, he managed to replicate my sample code. 
Apparently he ran into the exact same issue as me, where if you add 3 items to the list and submit the form then remove one item and re-submit, it'll show that you still have 3 items 😅

**Comment by Phibee** [comment, marked as answer]
@Skia69  I've tried to replicate your sample code and had it working on codesandbox. Please see the sample code below. Hope it helps.

### Code:
https://codesandbox.io/s/vibrant-panini-eeksq4?file=/src/App.js

**Reply by Skia69** [reply]
Thanks mate
